lib/modules/embeddings/topic.rb

```diff
@@ -33,55 +33,73 @@ def symmetric_semantic_search(model, topic)
       def asymmetric_semantic_search(model, query, limit, offset)
         embedding = model.generate_embedding(query)
 
-        candidate_ids =
-          DiscourseAi::Database::Connection
-            .db
-            .query(<<~SQL, query_embedding: embedding, limit: limit, offset: offset)
-          SELECT
-            topic_id
-          FROM
-            topic_embeddings_#{model.name.underscore}
-          ORDER BY
-            embedding #{model.pg_function} '[:query_embedding]'
-          LIMIT :limit
-          OFFSET :offset
-        SQL
-            .map(&:topic_id)
-
-        raise StandardError, "No embeddings found for topic #{topic.id}" if candidate_ids.empty?
+        begin
+          candidate_ids =
+            DiscourseAi::Database::Connection
+              .db
+              .query(<<~SQL, query_embedding: embedding, limit: limit, offset: offset)
+                SELECT
+                  topic_id
+                FROM
+                  topic_embeddings_#{model.name.underscore}
+                ORDER BY
+                  embedding #{model.pg_function} '[:query_embedding]'
+                LIMIT :limit
+                OFFSET :offset
+              SQL
+              .map(&:topic_id)
+        rescue PG::Error => e
+          raise StandardError, "No embeddings found for topic #{topic.id}" if candidate_ids.empty?
```

Comment:
I think this needs to be a `MissingEmbeddingError` so that it's caught upstream correctly?

https://github.com/discourse/discourse-ai/blob/b213fe7f94000205496ea5243c4801d4b3ebb252/lib/modules/embeddings/semantic_related.rb#L44

Would be great to add a spec which stubs `Database::Connection.db.query` to return a `PG::ConnectionBad`, and then makes an HTTP request. Then we'd have a nice end-to-end test of this handling.